qrenderdoc/Widgets/Extended/RDSplitter.cpp

```diff
@@ -0,0 +1,137 @@
+#include "RDSplitter.h"
+
+#include <QPainter>
+#include <QPaintEvent>
+#include <QApplication>
+
+
+RDSplitterDoubleClickEvent::RDSplitterDoubleClickEvent(int index)
+    :QEvent(staticType()), m_index(index)
+{
+
+}
+
+int RDSplitterDoubleClickEvent::index() const
+{
+    return m_index;
+}
+
+QEvent::Type RDSplitterDoubleClickEvent::staticType()
+{
+    static int type = registerEventType();
+    return (QEvent::Type)type;
+}
+
+RDSplitterHandle::RDSplitterHandle(Qt::Orientation orientation, QSplitter * parent)
+    :QSplitterHandle(orientation, parent), m_index(-1), m_isCollapsed(false)
+{
+
+}
+
+void RDSplitterHandle::setIndex(int index)
+{
+    m_index = index;
+}
+
+int RDSplitterHandle::index() const
+{
+    return m_index;
+}
+
+void RDSplitterHandle::setTitle(const QString &title)
+{
+    m_title = title;
+}
+
+const QString &RDSplitterHandle::title() const
+{
+    return m_title;
+}
+
+void RDSplitterHandle::setCollapsed(bool collapsed)
+{
+    m_isCollapsed = collapsed;
+}
+
+bool RDSplitterHandle::collapsed() const
+{
+    return m_isCollapsed;
+}
+
+void RDSplitterHandle::paintEvent(QPaintEvent * event)
+{
+    QPainter painter(this);
+    painter.setPen(Qt::black);
+
+    QString text(m_title);
+    if (m_isCollapsed) text.append(" ...");
+
+    painter.drawText(event->rect(), Qt::AlignCenter | Qt::TextWordWrap, text);
+}
+
+void RDSplitterHandle::mouseDoubleClickEvent(QMouseEvent *event)
+{
+    // post the event to the parent
+    RDSplitterDoubleClickEvent *rdEvent = new RDSplitterDoubleClickEvent(m_index);
+    QApplication::postEvent(parent(), rdEvent);
+}
+
+RDSplitter::RDSplitter(Qt::Orientation orientation, QWidget *parent)
+    :QSplitter(orientation, parent)
+{
+    initialize();
+}
+
+RDSplitter::RDSplitter(QWidget *parent)
+    :QSplitter(parent)
+{
+    initialize();
+}
+
+void RDSplitter::setHandleCollapsed(int pos, int index)
+{
+    QList<int> totalSizes = sizes();
+    RDSplitterHandle *rdHandle = static_cast<RDSplitterHandle *>(handle(index));
+    if (totalSizes[index] == 0) rdHandle->setCollapsed(true);
+    else rdHandle->setCollapsed(false);
+}
+
+void RDSplitter::initialize()
+{
+    connect(this, SIGNAL(splitterMoved(int,int)), this, SLOT(setHandleCollapsed(int, int)));
```

Comment:
I find the new connect syntax easier to read, so this can be written as:

`QObject::connect(this, &RDSplitter::splitterMoved, this, &RDSplitter::setHandleCollapsed);`